src/insights_client/metrics.py

```diff
@@ -90,13 +126,19 @@ def __init__(
             try:
                 auth_method = cfg.get("insights-client", "authmethod")
             except configparser.NoOptionError:
+                logger.debug("Metrics: authmethod not defined in insights-client config. Defaulting to CERT")
                 auth_method = AUTH_METHOD_CERT
 
             if auth_method == AUTH_METHOD_BASIC:
                 self.base_url = "cloud.redhat.com"
                 self.port = 443
-                u = cfg.get("insights-client", "username")
-                p = cfg.get("insights-client", "password")
+                try:
+                    u = cfg.get("insights-client", "username")
+                    p = cfg.get("insights-client", "password")
```

Comment:
Shouldn’t we be consistent with other configuration options and strip the values? I know that this may be tricky for credentials. We really don’t want to block out users that really have whitespace at the end of their passwords.

```suggestion
                    u = cfg.get("insights-client", "username").strip()
                    p = cfg.get("insights-client", "password").strip()
```